pkg/common/upgrade/managed.go

```diff
@@ -0,0 +1,273 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"github.com/Masterminds/semver"
+	"github.com/openshift/osde2e/pkg/common/cluster/healthchecks"
+	"github.com/spf13/viper"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"log"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	upgradev1alpha1 "github.com/openshift/managed-upgrade-operator/pkg/apis/upgrade/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/openshift/osde2e/pkg/common/config"
+	"github.com/openshift/osde2e/pkg/common/helper"
+)
+
+var (
+
+	// Namespace in which the managed-upgrade-operator runs
+	muoNamespace = "openshift-managed-upgrade-operator"
+	// default pod disruption budget timeout in minutes
+	muoPdbDrainTimeout = int32(5)
+	// the 'upgrade type' that the upgrade operator should use when upgrading
+	muoUpgradeType = upgradev1alpha1.OSD
+	// the name of the generated UpgradeConfig resource containing upgrade configuration
+	upgradeConfigName = "osde2e-upgrade-config"
+
+	// name of the workload for pod disruption budget tests
+	pdbWorkloadName = "pdb"
+	// directory containing pod disruption budget workload assets
+	pdbWorkloadDir = "/assets/workloads/e2e/pdb"
+
+	// data for managed-upgrade-operator's configmap, overriding defaults
+	configOverride = `maintenance:
+  controlPlaneTime: 90
+  workerNodeTime: 8
+scale:
+  timeOut: 15
+upgradeWindow:
+  timeOut: 60
+nodeDrain:
+  timeOut: 10
+healthCheck:
+  ignoredCriticals:
+  - DNSErrors05MinSRE
+  - MetricsClientSendFailingSRE
+  - UpgradeNodeScalingFailedSRE
+`
+)
+
+// TriggerManagedUpgrade initiates an upgrade using the managed-upgrade-operator
+func TriggerManagedUpgrade(h *helper.H) (*configv1.ClusterVersion, error) {
+	var cVersion *configv1.ClusterVersion
+	var err error
+	// setup Config client
+	cfgClient := h.Cfg()
+
+	// get current Version
+	getOpts := metav1.GetOptions{}
+	cVersion, err = cfgClient.ConfigV1().ClusterVersions().Get(context.TODO(), ClusterVersionName, getOpts)
+	if err != nil {
+		return cVersion, fmt.Errorf("couldn't get current ClusterVersion '%s': %v", ClusterVersionName, err)
+	}
+
+	image := viper.GetString(config.Upgrade.Image)
+	releaseName := viper.GetString(config.Upgrade.ReleaseName)
+
+	// determine requested upgrade targets
+	if image != "" {
+		return cVersion, fmt.Errorf("image-based managed upgrades are unsupported")
+	}
+
+	upgradeVersion := strings.Replace(releaseName, "openshift-v", "", -1)
```

Comment:
Try using [our utility function](https://godoc.org/github.com/openshift/osde2e/pkg/common/util#OpenshiftVersionToSemver) for this instead.

Reply:
Perfect - done in 2fbe0d2